In [15]:
import numpy as np
from math import factorial

def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Из колоды в 52 карты извлекаются случайным образом 4 карты.

#### a) Найти вероятность того, что все карты – крести. 

Всего в колоде 13 карт крести (как и любой другой масти)
Соответственно, нам нужно вычислить  количество сочетаний для выбора 4 карт из 13, и разделить его на количество всех возможных сочетаний.
Т.е.
$$C_{13}^4$$

разделить на

$$C_{52}^4$$

In [16]:
quant_13_4 = combinations(13,4)
quant_13_4

715

In [17]:
quant_52_4 = combinations(52,4)
quant_52_4

270725

In [18]:
prob = quant_13_4/quant_52_4
prob

0.0026410564225690276

#### Ответ: 0.0026410564225690276

#### б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

"Хотя бы один туз" значит, что нас устраивает любой случай, при котором в выборке будет присутствовать туз, при этом в ней может быть и 2 туза, и 3 туза, и 4 - нас устроит любой из этих исходов. 

Можно пойти от противного: определить вероятность того, что у нас в выборке не окажется туза и вычесть эту вероятность из 1

Количество вариантов, что ни одна из карт не будет тузом (тузов в колоде 4):

In [19]:
quant_48_4 = combinations(48,4)
quant_48_4

194580

Количество всех возможных исходов:

In [20]:
quant_52_4

270725

Вероятность, что в выборке не будет ни одного туза:

In [21]:
quant_48_4/quant_52_4

0.7187367254594146

Вероятность, что среди 4-х карт окажется хотя бы один туз

In [22]:
1 - quant_48_4/quant_52_4

0.2812632745405854

#### Ответ: 0.2812632745405854

### 2. На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9. Код содержит три цифры, которые нужно нажать одновременно. Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

Хитрость в том, что нажав одну кнопку, мы уже не сможем нажать её же повторно, соответственно, событие выбора произвольной цифры является зависимым

In [23]:
1/(combinations(10,1)*combinations(9,1)*combinations(8,1))

0.001388888888888889

#### Ответ: 0.001388888888888889 

### 3. В ящике имеется 15 деталей, из которых 9 окрашены. Рабочий случайным образом извлекает 3 детали. Какова вероятность того, что все извлеченные детали окрашены?

Найдем общее количество способов, которыми можно извлечь 3 произвольные детали:

In [24]:
quant_15_3 = combinations(15, 3)
quant_15_3

455

Найдем количество способов, которыми можно 100% извлечь окрашенные детали:

In [25]:
quant_9_3 = combinations(9, 3)
quant_9_3

84

Вероятность, что будут выбраны только окрашенные детали:

In [26]:
quant_9_3/quant_15_3

0.18461538461538463

#### Ответ: 0.18461538461538463

### 4. В лотерее 100 билетов. Из них 2 выигрышных. Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Сначала купим первый билет. Вероятность того, что он выигрышный 2/100 или 1/50

Далее купим второй билет, вероятность того, что он выигрышный - 1/99

Вероятность обоих совместных событий = (1/50)*(1/99)

In [27]:
tickets_2_from_100 = (1/50)*(1/99)
tickets_2_from_100

0.00020202020202020205

#### Ответ: 0.00020202020202020205